### DISCUSSION TITLE
Covariance matrix not supported in IES?

### DISCUSSION THREAD

**Discussion by d-edwards-gwmodeller** [usgs/pestpp#260] [Q&A] [answered]
For calibration using pilot points in layer 1, I am trying to reference covariance matrices in my pest control file after using PPCOV and ADDREG1 utilities. I have been advised to use the covariance matrices to guide parameter reasonableness based on spatial relationships between the points. The .mat files are cited alongside the names of regularisation observation groups.
I run into an issue with PESTPP-IES where I am given the error below:

_Error condition prevents further execution:
control file parsing error: observation covariance matrix detected for group 'REGUL_H_PP' - these are not supported...yet!_

This pest control file works when running PEST_HP_MKL with no issue. I've had a look through the gmdsi covmats tutorial document but haven't found anything that specifies any difference for referencing the cov matrices in the control file when using pestpp-ies as opposed to pest_hp_mkl. 

Is the use of the covariance matrices still advisable when using ies? 

Any help/advice would be appreciated. Thanks!

**Comment by jtwhite79** [comment, marked as answer]
If you want to supply a prior parameter covariance matrix for ies, it's done with the optional "parcov" argument.  Can you try that?

**Reply by d-edwards-gwmodeller** [reply]
Thanks j, that works. I did not realise this could be used to specify parameter covariance for history-matching as well as uncertainty!

**Reply by d-edwards-gwmodeller** [reply]
When supplying the cov matrix using parcov, I am now getting the following warning and pest does not move on to the second iteration:

![Untitled](https://github.com/usgs/pestpp/assets/140794832/ed31fede-d2bc-452d-bcdb-84989e50e01b)

I have not encountered this before and the number of observations with conflict is a large percentage of my calibration data set. Do you have any idea what might be causing prior data conflict? A single run using one of the sampled parameter sets returns reasonable simulated matches to the observation data (however there are 3 different observation group types included, with residuals of different scales - I have weighted these accordingly but am not sure I have done so correctly)

**Comment by jtwhite79** [comment]
That's just a warning - ies will normally continue...if you are using windows, it looks like you might have paused execution by clicking on the cmd window?

PDC is an interesting issue and it can only be detected with a prior Monte Carlo - essentially the Monte carlo simulation results don't cover the observed values. Since the prior ensemble represents maximum expected parameter uncertainty, if the prior Sim results don't cover the obs values, it might take some extreme parameters (or combinations) to force the results to reach those obs...

**Reply by d-edwards-gwmodeller** [reply]
At this point I've trialed multiple versions, adjusting some of the obs weights and other ies flag options - they return different numbers of conflicts but the issue persists and pestpp does not make it beyond the 36 run round. 

The issue of Monte carlo simulation results not covering the observed values could be true for one of the observation groups (known condition of zero flow rate, rather than measured observations - this has been included as an observed flow of 0.001 m3/d observation target however most of the simulated rates seem to exceed ~500) so I plan to trial a pestpp with this removed. What confused me is that the PDC occurs for the other observation groups as well...

Could it also be worth revising any of the ies settings such as ies_reg_factor(currently at 1.25) or specifying ies_accept_phi_fac(currently default)?
Thanks for the quick response again!

**Reply by jtwhite79** [reply]
Can you elaborate on the 36 run round?  I think that's the lambda testing runs with default settings? What happens after that?  What does the rec file say?  Ies shouldn't (hopefully) fail silently - there ought to be an error message.  It's possible that all the lambda test runs failed, esp if there are lots of big residuals from those conflicts - these can cause really large and extreme parameter changes.  

If you want to see the conflicts yourself, just load up the 0.obs.csv file and plot histograms of the conflicted columns and compare that to the obsval in the control file.  I think there is also a PDC summary csv file...

**Reply by d-edwards-gwmodeller** [reply]
NPHINORED was set to 4 and it was unable to find lamda/line search factor to meet ies_accept_phi - did not save the text from cmd but it did give the error message on termination. 

This is a sample from the pdc summary csv. Can I specify something within the control file which will increase the standard deviations used for each observed and simulated obs? I think that would solve the issue and could be what is wrong with my setup!

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/dedwards/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/dedwards/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">



observation | obs_mean | obs_std | obs_min | obs_max | obs_stat_min | obs_stat_max | sim_mean | sim_std | sim_min | sim_max | sim_stat_min | sim_stat_max | distance
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
obs_1 | 296.198 | 0.503393 | 295.209 | 297.636 | 296.702 | 295.695 | 318.426 | 2.39382 | 312.554 | 324.27 | 320.82 | 316.032 | 14.9177
obs_2 | 309.749 | 0.691595 | 307.388 | 311.423 | 310.441 | 309.058 | 299.02 | 0.93263 | 297.218 | 301.013 | 299.953 | 298.087 | 6.37589
obs_3 | 350.587 | 0.929343 | 348.821 | 352.918 | 351.516 | 349.658 | 310.07 | 0.985207 | 308.192 | 312.548 | 311.055 | 309.085 | 36.2729
obs_4 | 334.687 | 1.07121 | 332.795 | 337.683 | 335.758 | 333.616 | 311.517 | 0.989431 | 309.63 | 314.034 | 312.506 | 310.528 | 18.7616
obs_5 | 305.07 | 0.991386 | 302.368 | 307.921 | 306.061 | 304.078 | 314.11 | 1.71179 | 310.28 | 318.585 | 315.822 | 312.399 | 2.35924
obs_6 | 311.637 | 1.0425 | 309.598 | 314.344 | 312.679 | 310.594 | 319.578 | 2.09871 | 314.751 | 325.013 | 321.676 | 317.479 | 0.406606
obs_7 | 269.933 | 1.42985 | 266.702 | 274.044 | 271.363 | 268.504 | 321.951 | 2.94757 | 316.252 | 327.734 | 324.899 | 319.003 | 42.2072
obs_8 | 242.94 | 1.61151 | 239.35 | 246.982 | 244.552 | 241.329 | 263.603 | 12.202 | 247.803 | 288.503 | 275.805 | 251.401 | 0.821448
obs_9 | 291.903 | 0.953068 | 289.845 | 294.674 | 292.856 | 290.95 | 302.441 | 0.978325 | 300.526 | 304.161 | 303.42 | 301.463 | 5.85161
obs_10 | 303.715 | 1.18815 | 301.193 | 306.572 | 304.903 | 302.527 | 325.928 | 0.803403 | 324.319 | 327.954 | 326.732 | 325.125 | 17.747
obs_11 | 303 | 0.656126 | 301.296 | 304.422 | 303.657 | 302.344 | 279.616 | 0.780957 | 278.174 | 281.618 | 280.397 | 278.835 | 19.6779
obs_12 | 315.104 | 1.07321 | 312.793 | 317.748 | 316.177 | 314.031 | 322.56 | 0.492303 | 321.65 | 323.831 | 323.052 | 322.068 | 3.90162
obs_13 | 309.125 | 1.47233 | 305.484 | 312.843 | 310.598 | 307.653 | 321.563 | 2.32307 | 316.955 | 328.692 | 323.886 | 319.24 | 4.11182
ddn_1 | 16.2431 | 7.60322 | -2.47244 | 38.8488 | 23.8464 | 8.63992 | -4.16472 | 0.445967 | -5.0293 | -2.64032 | -3.7188 | -4.61069 | 0.167874
ddn_2 | 18.4692 | 6.95705 | -0.08564 | 35.6064 | 25.4263 | 11.5121 | -5.74651 | 0.465206 | -6.69205 | -4.14049 | -5.2813 | -6.21172 | 4.05485
ddn_3 | 17.265 | 6.94533 | -2.87795 | 34.4152 | 24.2104 | 10.3197 | -11.4611 | 0.499509 | -12.5247 | -9.79481 | -10.962 | -11.9606 | 6.91686
ddn_4 | 16.8547 | 7.70458 | -3.31887 | 38.0485 | 24.5593 | 9.15016 | -19.287 | 0.563964 | -20.5093 | -17.5384 | -18.723 | -19.851 | 14.2195
ddn_5 | 17.8481 | 6.48549 | 4.86154 | 32.4195 | 24.3336 | 11.3626 | -26.772 | 0.640827 | -28.1409 | -24.9457 | -26.131 | -27.4128 | 29.8072
ddn_6 | 20.2157 | 7.1603 | -2.82904 | 37.0913 | 27.376 | 13.0554 | -34.147 | 0.727851 | -35.6542 | -32.2205 | -33.419 | -34.8749 | 29.3915
ddn_7 | 18.5478 | 7.31273 | 2.52671 | 36.2121 | 25.8605 | 11.2351 | -44.8829 | 0.871449 | -46.5685 | -42.7437 | -44.011 | -45.7543 | 45.2704
ddn_8 | 19.3673 | 6.7702 | 4.43495 | 37.9324 | 26.1375 | 12.5971 | -46.3629 | 0.892582 | -48.0703 | -44.1928 | -45.47 | -47.2555 | 48.6278
flux_1 | -0.0788456 | 1.04237 | -2.20477 | 2.12826 | 0.963523 | -1.12121 | 472.139 | 290.502 | 13.17 | 988.53 | 762.641 | 181.638 | 11.0417
flux_2 | 0.0504411 | 1.08163 | -2.44218 | 2.65601 | 1.13207 | -1.03119 | 463.017 | 290.561 | 11.16 | 983.96 | 753.578 | 172.456 | 8.50399
flux_3 | -0.0105891 | 0.893686 | -1.9045 | 1.83997 | 0.883097 | -0.90428 | 471.877 | 292.154 | 11.73 | 987.74 | 764.032 | 179.723 | 9.89003
flux_4 | 0.192021 | 1.00921 | -2.27341 | 2.4489 | 1.20123 | -0.81719 | 466.919 | 291.661 | 9.44 | 986.8 | 758.581 | 175.258 | 6.9911
flux_5 | -0.031302 | 1.0051 | -2.23557 | 2.60278 | 0.973798 | -1.0364 | 469.416 | 293.084 | 9.25 | 988.79 | 762.5 | 176.332 | 6.64722
flux_6 | 0.0860426 | 1.18469 | -2.79801 | 3.78209 | 1.27074 | -1.09865 | 469.334 | 294.489 | 11.6 | 995.41 | 763.823 | 174.845 | 7.81791
flux_7 | 0.19243 | 0.8678 | -2.03527 | 2.47768 | 1.06023 | -0.67537 | 468.713 | 294.943 | 9.43 | 992.39 | 763.655 | 173.77 | 6.95232
flux_8 | 0.132071 | 0.90541 | -2.39803 | 2.70459 | 1.03748 | -0.77334 | 464.394 | 295.795 | 7.61 | 993.75 | 760.19 | 168.599 | 4.90541



</body>

</html>

**Reply by jtwhite79** [reply]
ok so ies didnt find a lambda+scale factor value that reduced phi acceptably for 4 testing cycles...usually with the GLM algorithm, if the expected search direction doesnt go down hill at all, this usually means something is goofed in the pest interface (at least for me!).  

The sim standard deviation is coming from the simulation results of the prior parameter ensemble - these are pretty narrow ranges!  So if you want that to be wider, you need to add more parameters, increase parameter bounds, or both...the obs standard deviation is derived from the weights unless you tell ies otherwise (with an optional "standard_deviation" column in the obs data section of the control file if you are using a version 2 control file...).  But I think the issue is with the lack of spread in the simulation results - just looking at `obs_2`, that says to me using maximum expected parameter variability, the variation in simulated results at location `obs_2` is only +/- 0.9 ft/meters.  Thats a pretty small range...
